### DISCUSSION TITLE
Rework the "node probing/detection"

### DISCUSSION THREAD

**Discussion by wmitsuda** [otterscan/otterscan#1324] [Ideas]
Background: when a page is loaded, we first try to connect to the configured Ethereum node, then try to identify if it has the Otterscan APIs enabled, then we present the page itself.

That was made this way in order to make it easier to diagnose common user misconfigurations and we were pretty successful on that (very few support questions, and when that happened, the error msgs were helpful in helping us to give proper advice to users).

However, that's not the optimal approach, because it runs a lot of network-related stuff during the first render of each browser tab and on every page refresh. It also seems redundant, if the user was able to successfully access Otterscan, there is not much need to run this kind of diagnostic again when opening an extra tab, or refreshing the page.

One first approach to minimize this overhead was implementing an opt-in flag to turn off those checks. This is already used in our hosted instances for sepolia and holesky because the Erigon nodes are hosted by us on public internet, so we assume they are properly configured :)

I think a better approach would be to:

- Totally remove this check on page render, so we eliminate this overhead completely. We should optimistically assume Otterscan is properly configured.
- If the node is not properly configure, it'll result in some error that should be captured by  some global error handler. That error handler should then run this diagnostic (probe node, check if ots API is enabled, etc, etc.).

Thoughts?

**Comment by wmitsuda** [comment]
Assuming we agree on the above solution, then I imagine the next steps should be (they can all be separate PRs in order to make it easier to review):

- Implement a global error capture mechanism that runs all current diagnostics and present to the user.
- Remove the up-front node checking. That would make page loading much faster (except for the chain discovery which needs to await for the first node connection and would block the rest of page render, but still I think it would be faster than it is today)
- Improvements to error detection screen, like #1280

**Comment by sealer3** [comment]
I agree, most of the time the connection is configured properly and the node is online, so the faster we can begin loading the target page, the better. Other than first-time configuration problems starting a new Otterscan instance, usually the only connection issues are when the Erigon node is down.

What is a "global" error capture mechanism? Would it capture errors only in the provider constructor?

**Reply by wmitsuda** [reply]
I was thinking about any error when trying to communicate with the node
